File: src/backend/Gerenciamento.Informacoes.ESocial/Gerenciamento.Informacoes.ESocial.Dominio/Entidades/Estagiario.cs
using Gerenciamento.Informacoes.ESocial.Dominio.Enums;
using Gerenciamento.Informacoes.ESocial.Dominio.ObjetosValor;

namespace Gerenciamento.Informacoes.ESocial.Dominio.Entidades;

public class Estagiario
{
    public int EstagiarioId { get; set; }
    public int TrabalhadorId { get; set; }
    public NaturezaEstagio NaturezaEstagio { get; set; }
    public AreaAtuacao AreaAtuacao { get; set; }
    public string? RazaoSocialInstEnsino { get; set; }
    public string? CnpjInstEnsino { get; set; }
    public string? NomeSupervisor { get; set; }
    public Endereco? EnderecoInstEnsino { get; set; }
    public Trabalhador? Trabalhador { get; set; }
}

File: src/frontend/gerenciamento-esocial-app/models/Estagiario.ts
// ====== Enums ======
export enum NaturezaEstagio {
  Obrigatoria = 0,
  NaoObrigatoria = 1,
}

export enum AreaAtuacao {
  Exatas = 0,
  Humanas = 1,
  CienciasMedicas = 2,
}

// ====== Interface ======
export interface Estagiario {
  estagiarioId: number;
  trabalhadorId: number;
  naturezaEstagio: NaturezaEstagio;
  areaAtuacao: AreaAtuacao;
  razaoSocialInstEnsino?: string;
  cnpjInstEnsino?: string;
  nomeSupervisor?: string;
}

// ====== Função de mapeamento do backend para TypeScript ======
export function mapEstagiarioFromApi(apiData: any): Estagiario {
  return {
    estagiarioId: apiData.EstagiarioId,
    trabalhadorId: apiData.TrabalhadorId,
    naturezaEstagio: apiData.NaturezaEstagio as NaturezaEstagio,
    areaAtuacao: apiData.AreaAtuacao as AreaAtuacao,
    razaoSocialInstEnsino: apiData.RazaoSocialInstEnsino ?? undefined,
    cnpjInstEnsino: apiData.CnpjInstEnsino ?? undefined,
    nomeSupervisor: apiData.NomeSupervisor ?? undefined,
  };
}